common/crypto/curl-p/hamming.c

```diff
@@ -1,47 +1,40 @@
 /*
- * Copyright (c) 2018 IOTA Stiftung
+ * Copyright (c) 2019 IOTA Stiftung
  * https://github.com/iotaledger/entangled
  *
  * Refer to the LICENSE file for licensing information
  */
 
-#include <stdio.h>
-
 #include "common/crypto/curl-p/hamming.h"
 #include "common/crypto/curl-p/ptrit.h"
 #include "common/crypto/curl-p/search.h"
 #include "common/crypto/curl-p/trit.h"
 
-short test(PCurl *const curl, unsigned short const security) {
-  unsigned short i, j, k;
-  signed short sum;
-  for (i = 0; i < sizeof(ptrit_s) * 8; i++) {
-    sum = 0;
-
-    for (j = 0; j < security; j++) {
-      for (k = j * HASH_LENGTH_TRYTE; k < (j + 1) * HASH_LENGTH_TRYTE; k++) {
-        if ((curl->state[k].low & (1uLL << i)) == 0) {
-          sum--;
-        } else if ((curl->state[k].high & (1uLL << i)) == 0) {
-          sum++;
-        }
-      }
+static test_result_t test(pcurl_t const *pcurl, test_arg_t security) {
+  size_t i;
+
+  for (i = 0; i < PTRIT_SIZE; i++) {
+    ptrit_t const *p = pcurl->state;
+    ssize_t sum = 0;
```

Comment:
Let's not introduce a new `ssize_t` ?

Reply:
Signed `size_t` was a good candidate to hold a "sum of trits". I guess `long` will also do.

Reply:
`long` it is. Done.